### PR TITLE
Add KalSeed selection

### PR DESCRIPTION
**Pull request by brownd1978**
This PR restructures REve access to KalSeeds, using the indirection the a art::Ptr<KalSeed> collection to allow more flexibility in defining which KalSeeds to view. It akso fixes some inefficiencies where KalSeeds were being copied instead of passed by reference.  It also adds EventID selection to the default path, resulting in simplification for the examples.

### CODE REVIEW

**Review by sophiemiddleton — APPROVED** [review]
I think this looks good, will merge once I check for the bug you are seeing on the Mu2e machines

**Comment by brownd1978** [comment]
One issue: with this change the KalSeed infobox no longer provides information on what kind of track fit this is: all tracks come from the same collection.  It's easy enough to fix, and I've already fixed it for the terminal dump printout, but it needs fixing before this should be merged.  I'll try to append a fix to this PR.

**Pull request merged**
